### DISCUSSION TITLE
[Question] How does actions/cache work on self hosted runners?

### DISCUSSION THREAD

**Discussion by domgreen** [community/community#160614] [Actions]
Hi quick question,
 
I'm running a self hosted running and using actions/cache and see it pulling down chunks of data:
```
Received 319053687 of 319053687 (100.0%), 5.8 MBs/sec
Cache Size: ~304 MB (319053687 B)
/usr/bin/tar -z -xf /home/build1/runner/_work/_temp/9f1ce8e5-9a89-47c4-ad2b-fd87c5174690/cache.tgz -P -C /home/build1/runner/_work/repo/repo
```
Where is it pulling the cache from (I take it somewhere in GitHub) and does this go towards any bandwidth usage?

**Comment by smorimoto** [comment]
Yes, it's pulled from the infrastructure managed by GitHub (Microsoft). Of course, that's true in terms of bandwidth usage between the storage server and the infrastructure you manage.

**Reply by domgreen** [reply]
Is this GitHub bandwidth usage? Where will it appear in the bill?

**Reply by smorimoto** [reply]
No, GitHub doesn't charge for that. I meant the infrastructure side that you are managing. If you run it on AWS, you will be billed for bandwidth usage by AWS. (Well, if an exciting idea like https://github.com/actions/cache/issues/354 is implemented, there is a possibility that in the future, there will be no bandwidth charges on the infrastructure side that we manage.)

**Comment by nebuk89** [comment]
Our cache is hosted in Azure :) but we don't charge for egress/ingress, nor does this count towards any of our bandwidth meters we have. As @smorimoto says, where ever you are hosting may charge you for the ingress/egress.
